registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -88,42 +92,73 @@
-
-		BiometricRecord biometricRecord = packetManagerService.getBiometricsByMappingJsonKey(registrationId,
-				MappingJsonConstants.INDIVIDUAL_BIOMETRICS, process,
-				ProviderStageName.CMD_VALIDATOR);
-		List<BIR> birs=biometricRecord.getSegments();
-		List<JSONObject> payloads=new ArrayList<>();
-		for(BIR bir: birs) {
-			if(bir.getOthers()!=null) {
-			for(Entry entry: bir.getOthers()) {
-				if(entry.getKey().equals("PAYLOAD")) {
-					payloads.add(new JSONObject(entry.getValue()));				
-				}
-			}
+			throws JsonProcessingException, IOException, BaseCheckedException, 
+				ApisResourceAccessException, JSONException {
+		List<String> fields = Arrays.asList(MappingJsonConstants.INDIVIDUAL_BIOMETRICS,
+				MappingJsonConstants.AUTHENTICATION_BIOMETRICS, 
+				MappingJsonConstants.INTRODUCER_BIO,
+				MappingJsonConstants.OFFICERBIOMETRICFILENAME, 
+				MappingJsonConstants.SUPERVISORBIOMETRICFILENAME);
+		for (String field : fields) {
+			String value = packetManagerService.getField(registrationId, field, process, 
+				ProviderStageName.PACKET_VALIDATOR);
+			if (value != null && !value.isEmpty()) {
+				BiometricRecord biometricRecord = packetManagerService.getBiometricsByMappingJsonKey(
+						registrationId, field, process,ProviderStageName.CMD_VALIDATOR);
+				if(biometricRecord == null)
+					throw new BaseCheckedException(
+						StatusUtil.DEVICE_VALIDATION_FAILED.getCode(),
+						StatusUtil.DEVICE_VALIDATION_FAILED.getMessage() + 
+							" --> Biometrics not found for field " + field);
+				validateDevicesInBiometricRecord(biometricRecord, regOsi);
 			}
 		}
-		if(payloads==null || payloads.isEmpty()) {
-			throw new BaseCheckedException(
-					StatusUtil.DEVICE_VALIDATION_FAILED.getCode(),StatusUtil.DEVICE_VALIDATION_FAILED.getMessage()+"-->Others info is not prsent in packet");
-		}
-		Set<String> signatures=new HashSet<>();
-		Set<String> deviceCodeTimestamps=new HashSet<>();
-		for(JSONObject payload :payloads) {
-				String digitalIdString=new String(CryptoUtil.decodeBase64(payload.getString("digitalId").split("\\.")[1]));
-				NewDigitalId newDigitalId=mapper.readValue(digitalIdString, NewDigitalId.class);
-				if(!signatures.contains(digitalIdString)) {
-					validateDigitalId(payload) ;
-					signatures.add(digitalIdString);
+	}
+
+	private void validateDevicesInBiometricRecord(BiometricRecord biometricRecord, RegOsiDto regOsi) 
+			throws JsonProcessingException, IOException, BaseCheckedException, 
+				ApisResourceAccessException, JSONException {
+		List<BIR> birs = biometricRecord.getSegments();
+		List<JSONObject> payloads = new ArrayList<>();
+		for(BIR bir : birs) {
+			if(bir.getOthers() != null) {
+				boolean exception = false;
+				String payload = "";
+				for(Entry entry: bir.getOthers()) {
+					if(entry.getKey().equals("EXCEPTION") && entry.getValue().equals("true") ) {
+						exception = true;
+						break;
```

Comment:
there could be only modality marked as exception right? in that case should other modalities and devices be validated?
Instead of break; should it be continue; in case of exception and in other cases it should add modalities in the payloads?